test/fake_pebble.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Fake (partial) Pebble server to allow testing the HTTP-over-Unix-socket protocol."""
+
+import http.server
+import json
+import os
+import re
+import socketserver
+import threading
+import urllib.parse
+
+
+class Handler(http.server.BaseHTTPRequestHandler):
```

Comment:
I didn't get the time to review the Handler yet.

Reply:
No worries -- can always adjust in a follow-up PR if you have further feedback.